managed-ledger/src/main/java/org/apache/bookkeeper/mledger/util/RangeCache.java

```diff
@@ -168,9 +170,11 @@ public Value get(Key key) {
             }
 
             Value value = entry.getValue();
-            ++removedEntries;
-            removedSize += weighter.getSize(value);
-            value.release();
+            long entrySize = weighter.getSize(value);
+            if (value.invalidate()) {
```

Comment:
I am afraid we are only hiding the problem.
there must be some coordination about these entries, some clear protocol about who is the owner of the entry.

when we get to this point then we must be sure that the refcount is valid on the value, otherwise it is always an hazard.

I believe that the right protocol is that  `before` calling` weighter.getSize(value);` we should `try` to acquire the entry and in case of failure we can ignore the entry.

```
Value value = entry.getValue();
if (value.tryAcquire()) {
            ++removedEntries;
            removedSize += weighter.getSize(value);
            value.release(); // this refers to the value.tryAcquire()
}
```

when we remove the entry we must have some write lock over the entry, that prevents double releases